tests/test_api.py

```diff
@@ -167,58 +167,263 @@ class TestApi(unittest.TestCase):
-                    'json_request_body': {
-                        "action": ["dss:CreateSubscription"],
-                        "resource": [f"arn:hca:dss:*:*:subscriptions/{email}/*"],
-                        "principal": "test@email.com"
-                    },
-                    'response': {
-                        'code': 200,
-                        'result': False
-                    }
-                },
-                {
-                    'json_request_body': {
-                        "action": ["fus:GetUser"],
-                        "resource": [f"arn:hca:fus:*:*:user/{email}/policy"],
-                        "principal": email
-                    },
-                    'response': {
-                        'code': 200,
-                        'result': True
-                    }
-                }
-            ]
-            for test in tests:
-                with self.subTest(test['json_request_body']):
-                    data=json.dumps(test['json_request_body'])
-                    headers={'Content-Type': "application/json"}
-                    headers.update(get_auth_header(service_accounts['admin']))
-                    resp = self.app.post('/v1/policies/evaluate', headers=headers, data=data)
-                    self.assertEqual(test['response']['code'], resp.status_code)  # TODO fix
-                    self.assertEqual(test['response']['result'], json.loads(resp.body)['result'])
-
-    @unittest.skip("incomplete")
-    def test_put_user(self):
-        pass
+        email = "test_evaluate_api@email.com"
+        tests = [
+            {
+                'json_request_body': {
+                    "action": ["dss:CreateSubscription"],
+                    "resource": [f"arn:hca:dss:*:*:subscriptions/{email}/*"],
+                    "principal": "test@email.com"
+                },
+                'response': {
+                    'code': 200,
+                    'result': False
+                }
+            },
+            {
+                'json_request_body': {
+                    "action": ["fus:GetUser"],
+                    "resource": [f"arn:hca:fus:*:*:user/{email}/policy"],
+                    "principal": email
+                },
+                'response': {
+                    'code': 200,
+                    'result': True
+                }
+            }
+        ]
+        for test in tests:
+            with self.subTest(test['json_request_body']):
+                data=json.dumps(test['json_request_body'])
+                headers={'Content-Type': "application/json"}
+                headers.update(get_auth_header(service_accounts['admin']))
+                resp = self.app.post('/v1/policies/evaluate', headers=headers, data=data)
+                self.assertEqual(test['response']['code'], resp.status_code)
+                self.assertEqual(test['response']['result'], json.loads(resp.body)['result'])
+
+    def test_put_new_user(self):
+        tests = [
```

Comment:
Please add a test with a user name that is > and == 64 characters.

Reply:
Added in the name tests

Reply:
move these tests to "test_user_api"

Reply:
done